packages/net/http/_test.pony

```diff
@@ -362,3 +367,196 @@ primitive _Test
     h.assert_eq[String](path, url.path)
     h.assert_eq[String](query, url.query)
     h.assert_eq[String](fragment, url.fragment)
+
+// Actor and classes to test the HTTPClient and modified _HTTPConnection.
+actor _HTTPConnTestHandlerActor
+  let h: TestHelper
+  var tries: USize
+  
+  new create(h': TestHelper, tries': USize) =>
+    h = h'
+    tries = tries'
+    h.log("_HTTPHandlerActor create called")
+
+  be apply(p: Payload val) =>
+    h.log("_HTTPHandlerActor apply called. Tries to go: " + tries.string())
+    if (tries = tries - 1) == 1 then 
+      h.complete(true)
+    end
+
+class _HTTPConnTestHandler is HTTPHandler
+  let h: TestHelper
+  let ha: _HTTPConnTestHandlerActor
+
+  new create(ha': _HTTPConnTestHandlerActor, h': TestHelper) =>
+    ha = ha'
+    h = h'
+    h.log("_HTTPConnTestHandler.create called")
+
+  fun ref apply(payload: Payload val): Any => 
+    h.log("_HTTPConnTestHandler.apply called")
+    ha(payload)
+
+  fun ref chunk(data: ByteSeq val) => 
+    h.log("_HTTPConnTestHandler.chunk called")
+
+class val _HTTPConnTestHandlerFactory is HandlerFactory
+  let h: TestHelper
+  let ha: _HTTPConnTestHandlerActor
+
+  new val create(ha': _HTTPConnTestHandlerActor, h': TestHelper) =>
+    ha = ha'
+    h = h'
+
+  fun apply(session: HTTPSession): HTTPHandler ref^ =>
+    h.log("_HTTPConnTestHandlerFactory.apply called")
```

Comment:
You could also call `TestHelper.dispose_when_done(session)` here instead of adding a `dispose()` to the `HTTPClient`, although i think `HTTPClient` should have such a method. But rather in another PR.